crates/net/ecies/src/algorithm.rs

```diff
@@ -36,23 +36,19 @@ fn ecdh_x(public_key: &PublicKey, secret_key: &SecretKey) -> B256 {
     B256::from_slice(&secp256k1::ecdh::shared_secret_point(public_key, secret_key)[..32])
 }
 
+/// This is the NIST SP 800-56A Concatenation Key Derivation Function (KDF) using SHA-256.
+///
+/// Internally this uses [`concat_kdf::derive_key_into`] to derive a key into the given `dest`
+/// slice.
+///
+/// # Panics
+/// * If the `dest` is empty
+/// * If `s1` is empty
```

Comment:
>  If `s1` is empty

I don't believe this is true. `s1` is allowed to be empty. Noting that this is a good thing as well call `kdf()` with and empty `s1`.